### DISCUSSION TITLE
Do char based extension for StartsWith and EndsWith make sense for `ReadOnlySpan<char>`

### DISCUSSION THREAD

**Discussion by SimonCropp** [dotnet/runtime#85374] [Ideas] [answered]
So [this issue](https://github.com/dotnet/runtime/issues/22434) managed the addition of 
```
 public static bool StartsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison);
 public static bool EndsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison);
```

but not the addition of 

```
 public static bool StartsWith(this ReadOnlySpan<char> span, char value);
 public static bool EndsWith(this ReadOnlySpan<char> span, char value);
```

Am i missing something that would mean these methods dont make sense for ` ReadOnlySpan<char>`?

**Comment by huoyaoyuan** [comment]
There are already overloads for generic T:
```csharp
    public static class MemoryExtensions
    {
        public static bool Contains<T>(this ReadOnlySpan<T> span, T value)
            where T: IEquatable<T>;
    }
```

Note the `IEquatabla<T>` constraint.

**Reply by SimonCropp** [reply]
ahh u r correct. so it is only StartsWith/EndsWith that are missing. i will update the discussion

**Comment by colejohnson66** [comment]
Probably because those can be simply implemented as `span.Length >= 1 && span[0] == value` (or `span[^1]` for `EndsWith`).

If such methods were to be added, I'd argue they should be generic over `T where T : IEquatable<T>` like `Contains` is. In fact, those string ones could probably do the same. There's no reason `StartsWith` (where `value` is a span) should be limited to characters.

**Comment by wstaelens** [comment]
which one should be preferred:

```
ros.Length > 0 && ros[0] == 'a';
```
or
```
ros.Length >= 1 && ros[0] == 'a';
```
or
```
!ros.IsEmpty && ros[0] == 'a';
```
?

**Reply by neon-sunset** [reply]
You can also write `ros is [.., 'a']` or `ros is ['a', ..]`.

**Comment by MihaZupan** [comment, marked as answer]
I didn't realize this discussion existed already, but here's an API proposal issue for span `StartsWith`/`EndsWith`: #87689
